packages/flutter/lib/src/foundation/_isolates_io.dart

```diff
@@ -45,8 +46,13 @@ Future<R> compute<Q, R>(isolates.ComputeCallback<Q, R> callback, Q message, { St
       result.completeError(exception, stack);
     }
   });
+  exitPort.listen((dynamic exitData) {
+    if (!result.isCompleted) {
+      result.completeError(Exception('Isolate exited without result or error.'));
```

Comment:
I'm not sure how this line could be reached because the spawned function will either return something or throw. The `dart:isolate` doc doesn't say too much about when and how onExit is used and interact with other ports.